File: HFSGuardaDiretorio_Java/hfsguardadirSWT/src/main/java/hfsguardadir/objetos/Extensao.java
package hfsguardadir.objetos;

import hfsguardadir.comum.Rotinas;
import java.util.Arrays;

public final class Extensao {

    private int codigo;
    private String nome;
    private int ordem;
    private byte[] bmp16;
    private byte[] bmp32;
    private byte[] gif16;
    private byte[] gif32;

    public Extensao() {
        this.limparDados();
    }

    public Extensao(int codigo, String nome, int ordem,
            byte[] bmp16, byte[] bmp32) {
        this.codigo = codigo;
        this.nome = nome;
        this.ordem = ordem;
        this.bmp16 = bmp16;
        this.bmp32 = bmp32;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getOrdem() {
        return ordem;
    }

    public void setOrdem(int ordem) {
        this.ordem = ordem;
    }

    public byte[] getBmp16() {
        return bmp16;
    }

    public void setBmp16(byte[] bmp16) {
        this.bmp16 = bmp16;
    }

    public byte[] getBmp32() {
        return bmp32;
    }

    public void setBmp32(byte[] bmp32) {
        this.bmp32 = bmp32;
    }

    /**
     * @return the gif16
     */
    public byte[] getGif16() {
        return gif16;
    }

    /**
     * @param gif16 the gif16 to set
     */
    public void setGif16(byte[] gif16) {
        this.gif16 = gif16;
    }

    /**
     * @return the gif32
     */
    public byte[] getGif32() {
        return gif32;
    }

    /**
     * @param gif32 the gif32 to set
     */
    public void setGif32(byte[] gif32) {
        this.gif32 = gif32;
    }

    public void limparDados() {
        this.codigo = 0;
        this.nome = "";
        this.ordem = 0;
        this.bmp16 = null;
        this.bmp32 = null;
    }

    public boolean equals(Extensao obj) {
        return (obj.getCodigo() == this.getCodigo()
                && obj.getNome().equals(this.getNome())
                && obj.getOrdem() == this.getOrdem());
    }

    @Override
    public String toString() {
        return Rotinas.getRecurso("Extensao.codigo") + " " + this.codigo + ", "
                + Rotinas.getRecurso("Extensao.nome") + " " + this.nome + ", "
                + Rotinas.getRecurso("Extensao.bmp16") + " "
                + Arrays.toString(this.bmp16) + ", "
                + Rotinas.getRecurso("Extensao.bmp32") + " "
                + Arrays.toString(this.bmp32);
    }
    
}

File: HFSGuardaDiretorio_Qt6/HFSGuardaDiretorio/hfsguardadir/objetos/Extensao.cpp
#include "Extensao.h"

namespace hfsguardadir {

namespace objetos {

Extensao::Extensao()
{
    this->limparDados();
}

Extensao::Extensao(int codigo, const QString &nome, int ordem,
	const QByteArray &bmp16, const QByteArray &bmp32) {
    this->codigo = codigo;
    this->nome = nome;
    this->ordem = ordem;
    this->bmp16 = bmp16;
    this->bmp32 = bmp32;
}

int Extensao::getCodigo() {
    return codigo;
}

void Extensao::setCodigo(int codigo) {
    this->codigo = codigo;
}

QString &Extensao::getNome() {
    return nome;
}

void Extensao::setNome(const QString &nome) {
    this->nome = nome;
}

int Extensao::getOrdem() {
    return ordem;
}

void Extensao::setOrdem(int ordem) {
    this->ordem = ordem;
}

QByteArray Extensao::getBmp16() {
    return bmp16;
}

void Extensao::setBmp16(const QByteArray &bmp16) {
    this->bmp16 = bmp16;
}

QByteArray Extensao::getBmp32() {
    return bmp32;
}

void Extensao::setBmp32(const QByteArray &bmp32) {
    this->bmp32 = bmp32;
}

QByteArray Extensao::getGif16() {
    return gif16;
}

void Extensao::setGif16(const QByteArray &gif16) {
    this->gif16 = gif16;
}

QByteArray Extensao::getGif32() {
    return gif32;
}

void Extensao::setGif32(const QByteArray &gif32) {
    this->gif32 = gif32;
}

void Extensao::limparDados() {
    this->codigo = 0;
    this->nome = "";
    this->ordem = 0;
    this->bmp16.clear();
    this->bmp32.clear();
    this->gif16.clear();
    this->gif32.clear();
}

bool Extensao::equals(Extensao *obj) {
    return (obj->getCodigo() == this->getCodigo()
            && obj->getNome() == this->getNome()
            && obj->getOrdem() == this->getOrdem());
}

QString &Extensao::toString() {
    return this->getNome();
}

}
}
